csrc/mmdeploy/preprocess/transform/pad.cpp

```diff
@@ -80,6 +85,20 @@ Result<Value> PadImpl::Process(const Value& input) {
       output["pad_size_divisor"] = arg_.size_divisor;
       output["pad_fixed_size"].push_back(pad_h);
       output["pad_fixed_size"].push_back(pad_w);
+    } else if (arg_.logical_or_val > 0) {
+      int pad_h = (height | arg_.logical_or_val) + arg_.add_pix_val;
+      int pad_w = (width | arg_.logical_or_val) + arg_.add_pix_val;
+      int offset_h = pad_h / 2 - height / 2;
+      int offset_w = pad_w / 2 - width / 2;
+      padding = {offset_w, offset_h, pad_w - width - offset_w, pad_h - height - offset_h};
+      output["border"].push_back(offset_h);
+      output["border"].push_back(offset_w);
+      output["border"].push_back(offset_h + height);
+      output["border"].push_back(offset_w + width);
+      if (!output.contains("scale_factor")) {
+        for (int i = 0; i < 4; ++i)
+          output["scale_factor"].push_back(1);  // initialize scale factor to 1 to avoid zero.
```

Comment:
Default scale_factor is handled in DefaultFormatBundle
https://github.com/open-mmlab/mmdeploy/blob/5923054bb417eb593e7543ed46d2980a40ef82b3/csrc/mmdeploy/preprocess/transform/default_format_bundle.cpp#L33

Semantically, Pad transform has nothing to do with "scale_factor".

Reply:
> Default scale_factor is handled in DefaultFormatBundle
> 
> https://github.com/open-mmlab/mmdeploy/blob/5923054bb417eb593e7543ed46d2980a40ef82b3/csrc/mmdeploy/preprocess/transform/default_format_bundle.cpp#L33
> 
> Semantically, Pad transform has nothing to do with "scale_factor".

`ResizeBBox::MapToOriginImage` needs a `scale_factor` with length 4, while the `scale_factor` generated from `default_format_bundle.cpp` only has length 1, which will lead the divide-0 bugs in `ResizeBBox::MapToOriginImage` function.
So, may be I should change the code in `default_format_bundle.cpp`?